### ISSUE TITLE
Numerical issue in water saturation routine

### ISSUE THREAD

**Issue by jranalli** [chmarti1/PYroMat#42]
Found a case where the numbers go a bit crazy. This is nowhere near the critical point (which is 220 bar). Expected value is around 592 K, which is also not especially near the critical point. 

```python
import pyromat as pm
h2o = pm.get('mp.H2O')
h2o.Ts(p=110) # yields nan
```

**Comment by chmarti1** [comment]
Confirmed.  This only occurs in a tight numerical range around 110bar.  I can recreate the failure for 110, 111, and 112 bar.  I get valid values everywhere else.

**Comment by chmarti1** [comment]
Confirmed.  The _Ts() algorithm allows iteration to wander 1% above critical and 1% below triple temperature.  The original thinking was that even if the underlying polynomial is meaningless outside of those bounds, it should still be continuous.  That approach is flawed, because the formulation uses fractional powers of a term that becomes negative above the critical point.  I can correct this issue by enforcing a strict limit against guesses outside the critical point.  In my tests, that works fine.

The only downside is that the saturation temperature will fail when _precisely_ the critical pressure is given.  Before I release the fix, I'll see if it makes sense to check for those values and coerce them to the known critical temperature.  I think that will work fine.

**Comment by chmarti1** [comment]
This issue is corrected in version 2.2.1.  I eliminated the 1% expansion on the iteration boundaries on the high side.  It completely mitigates the problem.  I have left the 1% expansion on the low side, so the triple point is still inside of the valid iteration domain even with numerical error.
